site-builder/src/main.rs

```diff
@@ -99,10 +99,22 @@ async fn run() -> Result<()> {
     tracing::info!(?config, "configuration loaded");
 
     match args.command {
-        Commands::Publish {
-            publish_options,
-            site_name,
-        } => {
+        Commands::Publish { publish_options } => {
+            let ws_res_path = publish_options
+                .clone()
+                .walrus_options
+                .ws_resources
+                .unwrap_or_else(|| {
+                    std::path::PathBuf::from(format!(
+                        "{}/ws-resources.json",
+                        publish_options.clone().directory.to_str().unwrap_or("")
+                    ))
+                });
+            let ws_resources = WSResources::read(ws_res_path);
+            let site_name = ws_resources.map_or_else(
+                |_| "test site".to_string(),
+                |res| res.name.unwrap_or_else(|| "test site".to_string()),
+            );
```

Comment:
Nit: just another way, up to your liking
```suggestion

            let site_name: String = ws_resources
                .ok()
                .and_then(|res| res.name)
                .unwrap_or("test site".to_string());
```

Reply:
Added this in a separate commits because the code has already changed.